Singularity/Singularity/screen/GameScreen.cs

```diff
@@ -0,0 +1,106 @@
+        /// <summary>
+        /// This list contains all the drawable objects currently in the game.
+        /// </summary>
+        private readonly LinkedList<IDraw> mDrawables;
+
+        /// <summary>
+        /// This list contains all the updateable objects currently in the game.
+        /// </summary>
+        private readonly LinkedList<IUpdate> mUpdateables;
+
+        // TODO: game screen needs the map in its constructor. Not in master as of now
+        public GameScreen()
+        {
+            mDrawables = new LinkedList<IDraw>();
+            mUpdateables = new LinkedList<IUpdate>();
+        }
+
+        public void Draw(SpriteBatch spriteBatch)
+        {
+            foreach(var drawable in mDrawables)
+            {
+                drawable.Draw(spriteBatch);
+            }
+        }
+
+        public bool DrawLower()
+        {
+            return false;
+        }
+
+        public void Update(GameTime gametime)
+        {
+            foreach (var updateable in mUpdateables)
+            {
+                updateable.Update(gametime);
+            }
+        }
+
+        public bool UpdateLower()
+        {
+            return false;
+        }
+
+        /// <summary>
+        /// Adds the given object to the game screens list of objects to handle.
+        /// </summary>
+        /// <typeparam name="T">The type of the object to be added. Needs to inherit from IDraw or IUpdate</typeparam>
+        /// <param name="toAdd">The object to be added to the game screen</param>
+        /// <returns>True if the given object could be added, false otherwise</returns>
+        public bool AddObject<T>(T toAdd)
+        {
+            if (!typeof(IDraw).IsAssignableFrom(typeof(T)) && !typeof(IUpdate).IsAssignableFrom(typeof(T)))
+            {
+                return false;
+            }
+
+            if (typeof(IDraw).IsAssignableFrom(typeof(T)))
+            {
+                mDrawables.AddLast((IDraw) toAdd);
+            }
+            if (typeof(IUpdate).IsAssignableFrom(typeof(T)))
+            { 
+                mUpdateables.AddLast((IUpdate) toAdd);
+            }
+            return true;
+
+        }
+
+        /// <summary>
+        /// Removes the given object from the game screens list of objects to handle.
+        /// </summary>
+        /// <typeparam name="T">The type of the object to be removed. Needs to inherit from IDraw or IUpdate</typeparam>
+        /// <param name="toRemove">The object to be removed to the game screen</param>
+        /// <returns>True if the given object could be removed, false otherwise</returns>
+        public bool RemoveObject<T>(T toRemove)
+        {
+            if (!typeof(IDraw).IsAssignableFrom(typeof(T)) && !typeof(IUpdate).IsAssignableFrom(typeof(T)))
+            {
+                return false;
+            }
+
+            if (typeof(IDraw).IsAssignableFrom(typeof(T)))
+            {
+                mDrawables.Remove((IDraw) toRemove);
+                return true;
```

Comment:
why do you already return true here? what if it's both drawable and updateable (can very well happen right?) - also, don't you need an IsAssignable if for that as well?

Reply:
Yeah, this ones my bad, i rewrote the code for the add method but forgot to for the remove method